src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -313,6 +313,10 @@ internal class MemberSignatureComparer : IEqualityComparer<Symbol>
             _considerCallingConvention = considerCallingConvention;
             _considerRefKindDifferences = considerRefKindDifferences;
             _typeComparison = typeComparison;
+            if (!considerRefKindDifferences)
+            {
+                _typeComparison |= TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly;
```

Comment:
>TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly [](start = 35, length = 57)

Can we assert that this option is never set on entry?

Reply:
We could, but I'm not sure what invariant that's would be trying to preserve? Could you elaborate on what type of bug you think that would catch?

---
In reply to: [426087571](https://github.com/dotnet/roslyn/pull/44257#discussion_r426087571) [](ancestors = 426087571)

Reply:
> We could, but I'm not sure what invariant that would be trying to preserve? Could you elaborate on what type of bug you think that would catch?

From reviewing the change, I assumed that ```TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly``` is a very special flag. It is meant to be used by this component and it is really not appropriate (suspicious) to use it anywhere else in the code base. If the assumption is correct, the assert would explicitly indicate the intent and could possibly prevent people from misusing the flag. 

---
In reply to: [426889306](https://github.com/dotnet/roslyn/pull/44257#discussion_r426889306) [](ancestors = 426889306,426087571)